### PR TITLE
Override core movement listeners

### PR DESCRIPTION
**Pull request by wvanlint**
This PR overrides any previous registered listeners to the core movement commands.
This is achieved by adding the autocompletion listener to the text editor element as the first listener as late as possible.
Depends on atom/atom#11506.
cc @nathansobo @as-cii

### CODE REVIEW

File: lib/suggestion-list.coffee [hunk 1]
```diff
@@ -190,12 +194,14 @@ class SuggestionList
     if marker = editor.getLastCursor()?.getMarker()
       @activeEditor = editor
       @overlayDecoration = editor.decorateMarker(marker, {type: 'overlay', item: this})
+      @bindToMovementCommands(editor)
```
Comment (as-cii) [inline lib/suggestion-list.coffee]: How would you feel about moving `@bindToMovementCommand`/`@movementCommandSubscriptions?.dispose()` into the `addKeyboardInteraction`/`removeKeyboardInteraction` methods? Seems exactly where those two pieces of behavior belong.

**Comment by as-cii** [comment]
One minor comment and then we can 🚢 this! 🎆

**Comment by wvanlint** [comment]
> How would you feel about moving @bindToMovementCommand/@movementCommandSubscriptions?.dispose() into the addKeyboardInteraction/removeKeyboardInteraction methods? Seems exactly where those two pieces of behavior belong.

Thanks for the review!
Sounds good! I wasn't sure due to keyboard bindings and command bindings being different.
I don't think separating the methods is necessary though, so I merged them and made the naming a bit more general.

**Comment by as-cii** [comment]
Thanks, @wvanlint! ✨

**Pull request merged**